### ISSUE TITLE
Date Tick Labels example

### ISSUE THREAD

**Issue by rcomer** [matplotlib/matplotlib#19203]
### Problem
The [Date Tick Labels](https://matplotlib.org/3.3.3/gallery/text_labels_and_annotations/date.html#sphx-glr-gallery-text-labels-and-annotations-date-py) example has a couple of lines whose purpose are unclear, and I'm actually not sure whether they are achieving anything:

https://github.com/matplotlib/matplotlib/blob/f2c5b37264a48eb7ecf7cc4924bdac7b85acd46f/examples/text_labels_and_annotations/date.py#L58-L60

I don't know what the "coords message box" referred to in the comment is.  I tried commenting out the two `ax.format_?data` lines, and the output plot seemed to be the same.

### Suggested Improvement

It looks like these lines have been there for a very long time so possibly they no longer do what they were originally written to do.  In which case they could just be removed.  It seems plausible that the intent of line 60 is to format the y-axis, so possibly that could be replaced with something like
```python
ax.yaxis.set_major_formatter(lambda x, pos: f'${x:.2f}')  # format the price.
```
If these lines are still doing something useful, then more information in that comment on line 58 would be really helpful.

**Comment by jklymak** [comment]
I suppose this could be a little more verbose, but this formats the numbers displayed as the pointer moves through the axes.  Of course you won't see this if you don't use an interactive backend.

**Comment by rcomer** [comment]
Thanks @jklymak.  I was using an interactive backend, but it had not occurred to me to wave my cursor over the plot!  This is functionality that I had no idea existed.

**Comment by rcomer** [comment]
I have opened #19204 to expand that comment.

**Comment by tacaswell** [comment]
Closed by #19204
